react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState({ isClicked: !this.state.isClicked });
+  }
+
+  render() {
+    return (this.state.isClicked ? <button onClick={this.handleClick}>Thanks!</button> : <button onClick={this.handleClick}>Click Me</button>
```

Comment:
Since just the text is changing it would be better to use the ternary to change the text:

```JSX
return <button onClick={this.handleClick}>{ this.state.isClicked ? 'Thanks!' : 'Click Me' }</button>;
```